### PR TITLE
3634-2

### PR DESCRIPTION
**Pull request by Mikitsik**
## Номер
3634

## Номер задания
2

## Ссылка на видео с демо
https://youtu.be/DGtiipD2uBo

### CODE REVIEW

File: 3634/2/commands/check_in.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'fileutils'
+require_relative 'save_files'
+
+module CheckInCommand
+  LONGITUDE = (53.914264..53.916233).freeze
```
Comment (anatoliliotych) [inline 3634/2/commands/check_in.rb]: move to module/class

File: 3634/2/commands/save_files.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class SaveFiles
+  attr_reader :session, :session_key, :payload
+
+  def initialize(session, session_key)
+    @session = session
+    @session_key = session_key
+  end
+
+  def make_path
+    FileUtils.makedirs path
```
Comment (anatoliliotych) [inline 3634/2/commands/save_files.rb]: try to use () it's more readable :)

File: 3634/2/commands/save_files.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def path
+    @path ||= "user/#{session_key}/#{Time.now.strftime '%Y-%m-%d_%H:%M:%S'}"
+  end
+
+  def save_files
+    make_path
```
Comment (anatoliliotych) [inline 3634/2/commands/save_files.rb]: How about making path in one method which will use existing or will create, but not splitting in two methods

File: 3634/2/commands/check_out.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    end
+  end
+
+  def selfie?
+    payload['photo']
+  end
+
+  def geoposition(*)
+    session[session_key]['location'] = payload['location']
+    # if geoposition?
```
Comment (anatoliliotych) [inline 3634/2/commands/check_out.rb]: why do you need this comment?

File: 3634/2/commands/check_in.rb (hunk 1 above)
Comment (anatoliliotych) [reply 3634/2/commands/check_in.rb]: you make lots of code duplication, try to inherit it from some abstract class

Comment (Mikitsik) [reply 3634/2/commands/check_in.rb]: I've removed code to module EventHandler (helpers/event_handler.rb) and reduce its duplication

File: 3634/2/commands/check_out.rb (hunk 4 above)
Comment (Mikitsik) [reply 3634/2/commands/check_out.rb]: I've removed it. It was just technickal remark

File: 3634/2/commands/save_files.rb (hunk 2 above)
Comment (Mikitsik) [reply 3634/2/commands/save_files.rb]: I've took this

File: 3634/2/commands/save_files.rb (hunk 3 above)
Comment (Mikitsik) [reply 3634/2/commands/save_files.rb]: Deleted make_path method and moved its inners to save_files method

**Comment by Mikitsik** [comment]
> where is env.sample file?

Added

File: 3634/2/data [hunk 5]
```diff
@@ -0,0 +1,38 @@
+3475
```
Comment (anatoliliotych) [inline 3634/2/data]: use yml/csv instead of unknown format!

Comment (Mikitsik) [reply 3634/2/data]: Done

File: 3634/2/helpers/event_handler.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      CAMPLATITUDE.include?(payload['location']['latitude'])
+  end
+
+  def start_save_files
+    respond_with :message, text: chat_session[session_key]['message']
+    status_switcher
+    SaveFiles.new(chat_session, session_key).save_files
+  end
+
+  def status_switcher
```
Comment (anatoliliotych) [inline 3634/2/helpers/event_handler.rb]: looks like hell
can you explain?

File: 3634/2/helpers/event_handler.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
+require 'fileutils'
+require_relative 'save_files'
+
+module EventHandler
+  CAMPLATITUDE = (53.913264..53.917233).freeze
```
Comment (anatoliliotych) [inline 3634/2/helpers/event_handler.rb]: CAMP_LATITUDE, try to use more human naming

File: 3634/2/helpers/event_handler.rb (hunk 6 above)
Comment (Mikitsik) [reply 3634/2/helpers/event_handler.rb]: When user do /checkin or /chekout commands he receive a specific status "in" or "out". For this opption if the user on "in" status he can not do /checkin command again. He is allowed  to do only the /checkout command and vise versa. 
The "status_switcher" method changes user's status if he've finished this specific command /checkin or /checkout

Comment (Mikitsik) [reply 3634/2/helpers/event_handler.rb]: When user is new to @workflow_checker_bot he has not any status. So he receive it in last stroke of this method

File: 3634/2/helpers/event_handler.rb (hunk 7 above)
Comment (Mikitsik) [reply 3634/2/helpers/event_handler.rb]: Done

**Pull request merged**